sycl/include/CL/sycl/detail/kernel_info.hpp

```diff
@@ -28,11 +28,11 @@ template <info::kernel Param> struct get_kernel_info_cl<string_class, Param> {
     if (ResultSize == 0) {
       return "";
     }
-    string_class Result(ResultSize, ' ');
+    vector_class<char> Result(ResultSize);
     // TODO catch an exception and put it to list of asynchronous exceptions
     CHECK_OCL_CODE(clGetKernelInfo(ClKernel, cl_kernel_info(Param), ResultSize,
-                                   &Result[0], nullptr));
-    return Result;
+                                   Result.data(), nullptr));
+    return string_class(Result.data());
```

Comment:
I have the feeling that just returning `Result` would have worked, isn't it?
Or perhaps with a final `.resize()` to skip the `\0`?
But anyway, it requires a comment about the "why"...

Reply:
I think there is more than one way to create a string from a vector:
std::string(vector.data());
std::string(vector.data().data(), vector.data().size());
std::string(vector.begin(), vector.end());
I don't want to rely on implicit conversion when I can explicitly specify it

At using resize I still have to do `ResultSize - 1` and explain it in the comments.

Do you think the current implementation needs comments?
As for me, it just uses only a clean API without any hacks.
If the code is not clear, I must edit it instead of adding comments.
Feel free to advice.

Reply:
Probably this can remains like this for now if it works.
More generally, the whole project needs comments, so do not feel targeted :-)
I opened more generally https://github.com/intel/llvm/issues/189
in the hope that the global comment issue  can be solved later...